lib/shelf-file.js

```diff
@@ -0,0 +1,112 @@
+"use strict";
+
+module.exports = (logger, requestPromise, ShelfMetadata, utils) => {
+    /**
+     * An object that represents a Shelf artifact.
```

Comment:
Good call. I think most of these were written up pretty quick. I'll take another look at them.